clang/include/clang/Basic/DiagnosticFrontendKinds.td

```diff
@@ -93,6 +93,8 @@ def err_fe_backend_error_attr :
 def warn_fe_backend_warning_attr :
   Warning<"call to '%0' declared with 'warning' attribute: %1">, BackendInfo,
   InGroup<BackendWarningAttributes>;
+def note_fe_backend_in : Note<"called by function '%0'">;
```

Comment:
Is this the right file for this?  It seems weird to do codegen in a 'frontend kinds' diagnostics list, though I'm not convinced I know enough to ask this question well enough.

Reply:
Where should the diagnostic strings go if not here?

Reply:
IDK, that is why I asked.  It SOUNDS wrong, so I'm hopeful someone else can come along and tell us.

Reply:
All of the diagnostics for backend warnings are defined in this file. (see existing lines 66-95).

As is, this note is emitted alongside the err/warn from existing lines 91-95, so it makes sense to put the new notes immediately after those existing definitions.

If my newly added backend warning notes belong in a new file, then all of the above belong in a new file.  But that seems orthogonal to my patch.